convert2rhel/checks.py

```diff
@@ -40,17 +42,50 @@ def perform_pre_checks():
 
 
 def check_uefi():
-    """Inhibit the conversion when UEFI detected."""
+    """Inhibit the conversion when we are not able to handle UEFI."""
     logger.task("Prepare: Checking the firmware interface type")
-    if os.path.exists("/sys/firmware/efi"):
+    if not grub.is_uefi():
         # NOTE(pstodulk): the check doesn't have to be valid for hybrid boot
-        # (e.g. AWS, Azure, OSP, ..)
+        # (e.g. AWS, Azure, ..)
+        logger.debug("BIOS detected")
+        return
+    logger.debug("UEFI detected.")
+    if not os.path.exists("/usr/sbin/efibootmgr"):
+        logger.critical("The UEFI has been detected but efibootmgr is not installed.")
+    if system_info.arch != "x86_64":
+        logger.critical("The conversion with UEFI is now handled only on the intel architecture.")
+    if grub.is_secure_boot():
+        # NOTE: need to be tested yet. So let's inhibit the conversion for now
+        # until we are sure it's safe...
+        logger.debug("Secure boot detected.")
+        logger.critical("The conversion with secure boot is currently not supported.")
+
+    # Load the data about the bootloader. Currently data is not used, but it's
+    # good check we can obtain all required data after the PONR. Better to
+    # stop now than later.
+    try:
+        efiboot_info = grub.EFIBootInfo()
+    except grub.BootloaderError as e:
+        # NOTE(pstodulk): maybe just print of the error msg could be enough..
```

Comment:
@shaded-enmity sorry, wrong wording :D The NOTE is that probably this could be enough:
```
logger.critical(e.message)
```

it always counts with the critical level -> inhibitor. WDYT? should I keep the msg as it is or just print the error message from the exception? I can remove the note then.

Reply:
ahh, makes sense. Yeah I'd just print `e.message` to make it more straightforward.

Reply:
Done.